Algorithms/Strings/PatternMatching/GeneralStringAlgorithms.cs

```diff
@@ -1,6 +1,6 @@
 using System;
 
-namespace Algorithms.Strings
+namespace Algorithms.Strings.PatternMatching
```

Comment:
This is not pattern matching. Could you just leave it in Strings? I can't come up with a better structure

Reply:
Internally it implements a LongestConsecutiveCharacter algorithm, which could be classified as Pattern Matching, but in general, yes, it is not very clear what or how this class can be organized. I moved it out of the Pattern Matching directory.